packages/api/src/controllers/asset.ts

```diff
@@ -105,6 +115,40 @@ function withPlaybackUrls(asset: WithID<Asset>, ingest: string): WithID<Asset> {
   };
 }
 
+async function reconcileAssetStorage(
+  { taskScheduler }: Request,
+  asset: WithID<Asset>,
+  newStorage: Asset["storage"],
+  task?: WithID<Task>
+): Promise<{ storage: Asset["storage"]; status: Asset["status"] }> {
+  let { storage, status } = asset;
+  const ipfsParamsEq =
+    JSON.stringify(newStorage.ipfs) === JSON.stringify(storage?.ipfs);
+  if (!ipfsParamsEq) {
+    if (!newStorage.ipfs) {
+      throw new BadRequestError("Cannot remove asset from IPFS");
+    }
+    if (!task) {
+      task = await taskScheduler.scheduleTask(
```

Comment:
So, here it spawns a new export task with the new IPFS metadata, if I understood correctly. I suppose the only thing you could change is the metadata CID as the video file CID would be always the same for the same asset, does this mean we are redoing the full export on the same CID reuploading the file?

Reply:
Yeah I agree, we should skip the re-importing of a same asset. I wonder if two different customers upload the same video file what happens. Of course we skip the re-import, but do we let any of them know the export didn't actually happen? I would say no, as on Livepeer they would have two separate assets in any case.

Reply:
I don't think we should expose cross-user information like that. It might execute a little faster but we won't expose why to the user. Also, that's another level of optimization that I hadn't thought of doing at first.

One thing is that we can avoid re-exporting the same asset multiple times. Another is to avoid exporting different assets with the same content. The latter would require us to actually compute the CID of an asset before it is exported to IPFS, then to check if we have already exported it from another asset by querying the DB for that CID. I think it's not a super important optimization to do either. It's a little more complex to implement and the user would not even be expecting anything like that anyway. It's also a scenario that will mostly happen on development and tests only, not on the real world with users ideally creating and minting NFTs with original content, so I'd say we shouldn't bother about that.